packages/design/src/DataTable/useTable.ts

```diff
@@ -200,6 +200,10 @@ function sortAndFilter<T>(
         const aValue = a[sort.key];
         const bValue = b[sort.key];
 
+        if (isISODate(aValue) && isISODate(bValue)) {
+          return new Date(aValue).getTime() - new Date(bValue).getTime();
+        }
+
```

Comment:
Thoughts on adding a test case for this to `Table.test.tsx`?

Reply:
Yeah, that's a good idea

Reply:
I see the new test... but, if I comment out your logic on lines 203-206, the tests still pass

Reply:
That's actually a pretty good catch. I had a mistake with my code. I assumed ISO dates are formatted in the same manner as  JavaScript does it. It's not true. Some are smaller even if they include a "Z" postfix. I have changed the code to remove that assumption and pasted explicit ISO strings as generated by our backend to properly test it.